tests/chainer_tests/distributions_tests/test_multivariate_normal.py

```diff
@@ -40,4 +45,72 @@ def sample_for_test(self):
         return smp
 
 
+@testing.parameterize(*testing.product({
+    'd': [3, 5],
+    'lower': [True, False],
+    'dtype': [numpy.float32],
+}))
+@testing.with_requires('scipy')
+class TestTriangularInv(unittest.TestCase):
+
+    def setUp(self):
+        self.x = numpy.random.normal(
+            0, 10, size=(self.d, self.d)).astype(self.dtype)
+        self.x = numpy.tril(self.x)
+        if not self.lower:
+            self.x = self.x.T
+        self.gy = numpy.random.normal(size=(self.d, self.d)).astype(self.dtype)
+        self.ggy = numpy.random.normal(
+            size=(self.d, self.d)).astype(self.dtype)
+        self.backward_options = {'atol': 1e-2, 'rtol': 1e-2}
+        self.double_backward_options = {'atol': 1e-2, 'rtol': 1e-2}
+
+    def check_forward(self, x_data):
+        xp = cuda.get_array_module(x_data)
+        y = distributions.multivariate_normal._triangular_inv(
+            x_data, lower=self.lower)
+        if xp == cuda.cupy:
+            y_xp = xp.asarray(numpy.linalg.inv(x_data.get()))
+        else:
+            y_xp = numpy.linalg.inv(x_data)
+        testing.assert_allclose(y.array, y_xp)
+
+    def test_forward_cpu(self):
+        self.check_forward(self.x)
+
+    @attr.gpu
+    def test_forward_gpu(self):
+        self.check_forward(cuda.to_gpu(self.x))
+
+    def check_backward(self, x_data, y_grad):
+        def f(x):
+            return distributions.multivariate_normal._triangular_inv(
+                x, lower=self.lower)
+        gradient_check.check_backward(
+            f, x_data, y_grad, **self.backward_options)
+
+    def test_backward_cpu(self):
+        self.check_backward(self.x, self.gy)
+
+    @attr.gpu
+    def test_backward_gpu(self):
+        self.check_backward(cuda.to_gpu(self.x), cuda.to_gpu(self.gy))
+
+    def check_double_backward(self, x_data, y_grad, x_grad_grad):
+        def f(x):
+            return distributions.multivariate_normal._triangular_inv(
+                x, lower=self.lower)
+        gradient_check.check_double_backward(
+            f, x_data, y_grad, x_grad_grad,
+            **self.double_backward_options)
+
+    def test_double_backward_cpu(self):
+        self.check_double_backward(self.x, self.gy, self.ggy)
+
+    @attr.gpu
+    def test_double_backward_gpu(self):
+        self.check_double_backward(
+            cuda.to_gpu(self.x), cuda.to_gpu(self.gy), cuda.to_gpu(self.ggy))
+
+
```

Comment:
Could you add tests `without_requires('scipy')`, as in `tests/chainer_tests/functions_tests/math_tests/test_lgamma.py`?